icalevents/icalparser.py

```diff
@@ -151,16 +150,14 @@ def create_event(component):
     :param component: iCal component
     :return: event
     """
-    event_start = normalize(component.get('dtstart').dt)
-    event_end = normalize(component.get('dtend').dt)
 
     event = Event()
 
-    event.start = event_start
-    event.end = event_end
+    event.start = normalize(component.get('dtstart').dt)
+    event.end = normalize(component.get('dtend').dt)
     event.summary = str(component.get('summary'))
-    event.description  = str(component.get('description'))
-    event.all_day = type(component.get('dtstart').dt) is date
+    event.description = str(component.get('description'))
+    event.all_day = isinstance(component.get('dtstart').dt, date)
```

Comment:
Are you sure about that check? Non-all-day events have `datetime` objects here, which inherit from `date`, i.e. they are `date` instances. This would set `all_day = True` for every event.

Reply:
Good catch. I have not tested it but I see this is a good reason to keep the previous way.